plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/actions/datasource/SelectActiveDataSourceHandler.java

```diff
@@ -76,6 +77,8 @@ public Object execute(ExecutionEvent event) throws ExecutionException
                 IFile curFile = EditorUtils.getFileFromInput(activeEditor.getEditorInput());
                 if (curFile != null) {
                     activeProject = DBPPlatformDesktop.getInstance().getWorkspace().getProject(curFile.getProject());
+                } else {
+                    activeProject = DBWorkbench.getPlatform().getWorkspace().getActiveProject();
```

Comment:
And what if it is not the current project? Hm...

Reply:
The script having this NPE doesn't have data source or project. So, I don't understand the question.

The idea is to return back this fallback, removed while refactoring, to prevent NPE.

Reply:
Ok